### DISCUSSION TITLE
Cancelling an intent

### DISCUSSION THREAD

**Discussion by realsteel1** [finos/FDC3#1254] [General]
Hello! 

What is the recommended way to cancel an intent that is 'in-flight'?

I appreciate there are implications with regard to the atomicity of an operation that could be in progress on the listener side, but I'd be interested to hear of any suggestions or existing recommended approaches for this.

Thank you,
Phil

**Comment by kriswest** [comment]
To my knowledge, the only use case I've seen for cancelling an intent is via an intent resolver UI. Intents would usually be delivered quickly (near instantly) if there is only a single option for resolution or a target has been chosen. If the target is ambiguous (there are multiple options) an intent resolver will usually be shown by a Desktop Agent, which may include a cancel button (there is one in my firm's implementations for example). If the user hits that the Desktop Agent can reject the `raiseIntent` promise with [`ResolveError.UserCancelled](https://fdc3.finos.org/docs/api/ref/Errors#resolveerror) to indicate that it happened.

Does that answer the question? If not please provide more detailed on your use case and desired behavior.
